pkg/controller/direct/redis/cluster/roundtrip_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/test/fuzz"
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/encoding/prototext"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/testing/protocmp"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	pb "cloud.google.com/go/redis/cluster/apiv1/clusterpb"
+)
+
+func FuzzRedisClusterSpec(f *testing.F) {
+	f.Fuzz(func(t *testing.T, seed int64) {
+		randStream := rand.New(rand.NewSource(seed))
+
+		p1 := &pb.Cluster{}
+		fuzz.FillWithRandom(t, randStream, p1)
+
+		// We don't expect output fields to round-trip
+		outputFields := sets.New(".etag")
+
+		// A few fields are not implemented yet in KRM, don't test them
+		unimplementedFields := sets.New(
+			".name",
+			".labels",
+		)
+
+		// Status fields
+		unimplementedFields.Insert(".discovery_endpoints")
```

Comment:
This file is not currently generated.  I think it would be nice to collect more examples and then automate.  e.g. I _think_ we can tell which fields are spec and which are status by seeing which ones map to spec and which to status (and if they map to neither, _that's_ an error)